lib/components/base-components/PrimitiveComponent/cssSelectPrimitiveComponentAdapter.ts

```diff
@@ -40,6 +41,33 @@ export const cssSelectPrimitiveComponentAdapter: Required<
           ? String(value)
           : null
     }
+
+    // Check for properties directly on the node
+    if (name in node) {
+      const value = (node as any)[name]
+      return typeof value === "string"
+        ? value
+        : value !== null && value !== undefined
+          ? String(value)
+          : null
+    }
+
+    // Use pre-computed attribute mapping for fast camelCase->lowercase lookups
+    if ("_attributeCamelToLowerNameMap" in node) {
+      const attrMap = (node as any)._attributeCamelToLowerNameMap
+      // Find the camelCase property that maps to this lowercase name
+      for (const [camelCaseName, lowerCaseName] of Object.entries(attrMap)) {
```

Comment:
do not iterate, you don't need to, it looks like you need the reverse map `_attributeLowerToCamelNameMap`